gradle/config.groovy

```diff
@@ -0,0 +1,3 @@
+withConfig(configuration) {
```

Comment:
I'm not exactly sure what this does, but [here](https://stackoverflow.com/a/38620029/4478636) is another way of doing the same as what you've done in `common.gradle` and `gradle/config.groovy`. I think?

Reply:
Right, that link shows both methods of adding the compiler configuration script elements required.  I believe that the option to put them in a single file is newer.

For some reason a single file containing all the compiler-directives appealed to me, and I found the AST version easier to read.  Do you have a strong-preference for the scripted version, eliminating `config.groovy`?

Reply:
No strong preference at all. :) Just less code/files but you are right about having all configurations in a single place. So leave it like that.

Reply:
It's probably moot anyway based on the discussion over in the original issue ... making everything `@CompileStatic` is being discussed as overkill.  Apparently the only thing that will make any difference is Gradle configuration-time activity, when the plugin is applied, not when it executes.

Reply:
Well, in our project we have a lot of Groovy and at some point I decided I hated the performance and went ahead and added `CompileStatic` *everywhere* and it ran **much** faster. I don't see any reason for not adding it absolutely everywhere (apart where heavy meta-programming or reflection is used) even if that would make only the compile-time faster (which is a win again).

Reply:
> and went ahead and added CompileStatic everywhere

Interesting - does that mean that your project was pretty-much Java-code but compiled with Groovy?  Based on my first experience here trying to add `@CompileStatic` in Gretty - pretty much any syntax which is not strict Java or anything with any dynamic typing at all immediately fails the static-compile, unsurprisingly.  Each of the files I've had to exclude should have at least one example of something that can only be done with Groovy dynamic allowances.

I'm guessing it's not going to be helpful or fun converting a bunch of those to work in static Java compilation, even for the ones that are "reasonable" changes.

Reply:
We have mostly Java code with Groovy for tests and some other minor stuff - so it wasn't too much non-Java code. Yes, we had to change the Groovy (perhaps even a lot in some places - but mostly adding type annotations) but in the end it was worth it. I'm not sure here it will be - probably it will be too much work, not sure.

I guess what you have done here suffices for now. Although I'm not exactly sure how the Groovy compiler can generate "fast" bytecode when it doesn't know the types... :)